plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcClient.java

```diff
@@ -253,6 +254,15 @@ default Optional<TableScanRedirectApplicationResult> getTableScanRedirection(Con
 
     OptionalInt getMaxWriteParallelism(ConnectorSession session);
 
+    JdbcTableHandle updatedScanColumnsForMerge(ConnectorSession session, ConnectorTableHandle table, Optional<List<JdbcColumnHandle>> originalColumns, JdbcColumnHandle mergeRowIdColumnHandle);
+
+    String buildMergeRowIdConjuncts(ConnectorSession session, List<String> mergeRowIdFieldNames, List<Type> mergeRowIdFieldTypes);
+
+    default List<JdbcColumnHandle> getPrimaryKeys(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        return List.of();
```

Comment:
Also should this information be "cached" in the table handle or column handles? It's not cheap to fetch it since requires remote I/O and we call this multiple times during a single merge query.